File: apnaghar-frontend/src/services/reviewService.js
import api from "./api";

export const getRatingSummary = (propertyId) => {
  return api.get(`/reviews/property/${propertyId}/summary`);
};

export const getReviewsByProperty = (propertyId) => {
  return api.get(`/reviews/property/${propertyId}`);
};

export const addReview = (data) => {
  return api.post("/reviews", null, { params: data });
};

File: src/main/java/com/apnaghar/service/ReviewService.java
package com.apnaghar.service;

import java.util.List;

import com.apnaghar.dto.RatingSummaryResponse;
import com.apnaghar.model.Review;

public interface ReviewService {

    Review addReview(Long propertyId, int rating, String comment, String userEmail);

    List<Review> getReviewsByProperty(Long propertyId);

    // ⭐ Rating summary
    RatingSummaryResponse getRatingSummary(Long propertyId);
}
